lib/runtime/storage/storagediff.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package storage
+
+import (
+	"bytes"
+	"maps"
+	"sync"
+
+	xmaps "golang.org/x/exp/maps"
+
+	"github.com/ChainSafe/gossamer/pkg/trie"
+	"github.com/ChainSafe/gossamer/pkg/trie/codec"
+)
+
+type storageDiff struct {
+	upserts        map[string][]byte
+	deletes        map[string]bool
+	childChangeSet map[string]*storageDiff
+	l              sync.RWMutex
```

Comment:
nit/either embed or rename to something like "mtx" or "lock"

Reply:
Done